components/application-connector/internal/metadata/servicedefservice.go

```diff
@@ -259,3 +274,37 @@ func (sds *serviceDefinitionService) insertSpecs(id string, docs []byte, api *se
 
 	return sds.minioService.Put(id, documentation, apiSpec, eventsSpec)
 }
+
+func tryToModifyAPISpec(rawApiSpec []byte, gatewayUrl string) ([]byte, apperrors.AppError) {
```

Comment:
I think we could refactor it a little bit. My proposal would be as follows:
- one function updating Host, BasePath ans Schemes properties (e.g. setBaseURL or similar)
- one function unmarshaling API spec, calling setBaseURL and marshaling back